source/android/adaptivecards/src/main/java/com/microsoft/adaptivecards/renderer/readonly/ImageRenderer.java

```diff
@@ -77,27 +85,29 @@ protected Bitmap doInBackground(String... args)
                     bitmap = circleBitmap;
                 }
 
-                return bitmap;
+                if (bitmap == null)
+                {
+                    throw new IOException("Failed to convert content to bitmap: " + new String(bytes));
+                }
```

Comment:
The null check in 77 and this is different.  77 checks to ensure BitmapFactory.decodeByteArray is not null while this is checking to ensure Bitmap.createBitmap does not return null.  Granted, new Canvas should verify, but this is just a safety check.

---
In reply to: [126748644](https://github.com/Microsoft/AdaptiveCards/pull/461#discussion_r126748644) [](ancestors = 126748644)